src/ImuSensor_TEST.cc

```diff
@@ -511,6 +516,304 @@ TEST(ImuSensor_TEST, CustomRpyParentFrame)
+
+  // Case C.1 : Localization ref: NED, World : NED
+  // Sensor aligns with NED, we're measuring wrt NED
+  {
+    const math::Quaterniond worldFrameOrientation(0, 0, 0);
+    const sensors::WorldFrameEnumType worldRelativeTo =
+      sensors::WorldFrameEnumType::NED;
+    const math::Quaterniond expectedSensorOrientation(0, 0, 0);
+
+    sensorNED->SetWorldFrameOrientation(worldFrameOrientation,
+      worldRelativeTo);
+    sensorNED->Update(std::chrono::steady_clock::duration(
+      std::chrono::nanoseconds(10000000)));
+    EXPECT_EQ(expectedSensorOrientation, sensorNED->Orientation());
+  }
+
+  // Case C.2 : Localization ref: NED, World : NED + rotation offset
+  {
+    const math::Quaterniond worldFrameOrientation(0, 0, IGN_PI/4);
+    const sensors::WorldFrameEnumType worldRelativeTo =
+      sensors::WorldFrameEnumType::NED;
+    const math::Quaterniond expectedSensorOrientation(0, 0, -IGN_PI/4);
+
+    sensorNED->SetWorldFrameOrientation(worldFrameOrientation,
+      worldRelativeTo);
+    sensorNED->Update(std::chrono::steady_clock::duration(
+      std::chrono::nanoseconds(10000000)));
+    EXPECT_EQ(expectedSensorOrientation, sensorNED->Orientation());
+  }
+
+  // Case C.3 : Localization ref: NED, World : NWU
+  // Sensor aligns with NWU, we're measuring wrt NED
+  {
+    const math::Quaterniond worldFrameOrientation(0, 0, 0);
+    const sensors::WorldFrameEnumType worldRelativeTo =
+      sensors::WorldFrameEnumType::NWU;
+    const math::Quaterniond expectedSensorOrientation(-IGN_PI, 0, 0);
+
+    sensorNED->SetWorldFrameOrientation(worldFrameOrientation,
+      worldRelativeTo);
+    sensorNED->Update(std::chrono::steady_clock::duration(
+      std::chrono::nanoseconds(10000000)));
+    EXPECT_EQ(expectedSensorOrientation, sensorNED->Orientation());
+  }
+
+  // Case C.4 : Localization ref: NED, World : ENU
+  // Sensor aligns with ENU, we're measuring wrt NED
+  {
+    const math::Quaterniond worldFrameOrientation(0, 0, 0);
+    const sensors::WorldFrameEnumType worldRelativeTo =
+      sensors::WorldFrameEnumType::ENU;
+    const math::Quaterniond expectedSensorOrientation(-IGN_PI, 0, IGN_PI/2);
+
+    sensorNED->SetWorldFrameOrientation(worldFrameOrientation,
+      worldRelativeTo);
+    sensorNED->Update(std::chrono::steady_clock::duration(
+      std::chrono::nanoseconds(10000000)));
+    EXPECT_EQ(expectedSensorOrientation, sensorNED->Orientation());
+  }
+}
+
+//////////////////////////////////////////////////
+TEST(ImuSensor_TEST, LocalizationTagEmpty)
+{
+  // Create a sensor manager
+  ignition::sensors::Manager mgr;
+
+  sdf::ElementPtr imuSDF;
+
+  std::ostringstream stream;
+  stream
+    << "<?xml version='1.0'?>"
+    << "<sdf version='1.6'>"
+    << " <model name='m1'>"
+    << "  <link name='link1'>"
+    << "    <sensor name='imu_sensor' type='imu'>"
+    << "      <topic>imu_test</topic>"
+    << "      <update_rate>1</update_rate>"
+    << "      <imu>"
+    << "      <orientation_reference_frame>"
```

Comment:
I think we should add `<localization>UNRECOGNIZED</localization>` or something in order to reach that line of code mentioned above

Reply:
Done